avail.go

```diff
@@ -0,0 +1,157 @@
+package avail
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/rollkit/go-da"
+)
+
+type SubmitRequest struct {
+	Data string `json:"data"`
+}
+
+type SubmitResponse struct {
+	BlockHash        string `json:"block_hash"`
+	TransactionHash  string `json:"hash"`
+	TransactionIndex uint32 `json:"index"`
+}
+
+type BlocksResponse struct {
+	BlockNumber      uint32             `json:"block_number"`
+	DataTransactions []DataTransactions `json:"data_transactions"`
+}
+
+type DataTransactions struct {
+	Data      string `json:"data"`
+	Extrinsic string `json:"extrinsic"`
+}
+
+type Config struct {
+	LcURL     string `json:"lc_url"`
+	BlocksURL string `json:"blocks_url"`
+}
+
+// AvailDA implements the avail backend for the DA interface
+type AvailDA struct {
+	appID  uint32
+	config Config
+	ctx    context.Context
+}
+
+// NewAvailDA returns an instance of AvailDA
+func NewAvailDA(appID uint32, ctx context.Context) *AvailDA {
+	return &AvailDA{
+		appID:  appID,
+		ctx:    ctx,
+		config: Config{LcURL: "http://localhost:8000/v2", BlocksURL: "/blocks/%d/data?fields=data,extrinsic"},
+	}
+}
+
+var _ da.DA = &AvailDA{}
+
+// submits each blob to avail data availability layer
+func (c *AvailDA) Submit(daBlobs []da.Blob) ([]da.ID, []da.Proof, error) {
+	ids := make([]da.ID, len(daBlobs))
+	for index, blob := range daBlobs {
+		encodedBlob := base64.StdEncoding.EncodeToString(blob)
+		requestData := SubmitRequest{
+			Data: encodedBlob,
+		}
+		requestBody, err := json.Marshal(requestData)
+		if err != nil {
+			return nil, nil, err
+		}
+		// Make a POST request to the /v2/submit endpoint.
+		response, err := http.Post(c.config.LcURL+"/submit", "application/json", bytes.NewBuffer(requestBody))
+		if err != nil {
+			return nil, nil, err
+		}
+		defer response.Body.Close()
+
+		responseData, err := io.ReadAll(response.Body)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		var submitResponse SubmitResponse
+		err = json.Unmarshal(responseData, &submitResponse)
+		if err != nil {
+			return nil, nil, err
+		}
+		ids[index] = makeID(submitResponse.TransactionIndex, submitResponse.BlockHash)
```

Comment:
If I'm not wrong, this will result in having different IDs returned by submit and by getIDs. Can we use the same IDs across the entire implementation? GetIDs would have to fetch the header to access the block hash, and also to fetch the data to access the indexes, but that would be only for this iteration until we extend the API?

Reply:
yes you're right @aterentic-ethernal , we can get hash when we fetch the header to access block hash but how can we access indexes by fetching the data as it's response field  contains only hash and data transactions

Reply:
Thats true, indexes on submit and data are not the same. Ill check if we can do something here.

Reply:
@aterentic-ethernal header endpoint also consists of `index` field, what do you thinking of fetching both block hash and index from that endpoint?

Reply:
I have missed this one :/ I think that makes sense, but it complicates adapter code, and that can be done on the LC side, since it has access to the same headers. We are thinking of indexing all headers to avoid additional RPC call on the submit endpoint, but thats work in progress.